src/rules/requireExample.js

```diff
@@ -1,23 +1,30 @@
 import _ from 'lodash';
 import iterateJsdoc from '../iterateJsdoc';
+import warnRemovedSettings from '../warnRemovedSettings';
 
 export default iterateJsdoc(({
   jsdoc,
   report,
   utils,
-  settings
+  context
 }) => {
+  warnRemovedSettings(context, 'require-example');
+
   if (utils.avoidDocs()) {
     return;
   }
 
+  const options = context.options[0] || {
+    avoidExampleOnConstructors: false
+  };
```

Comment:
I might suggest something like this here instead

```js
  const {avoidExampleOnConstructors = false} = context.options[0] || {};
```

Besides being potentially shorter, this allows for the `avoidExampleOnConstructors` default to be applied in the future if other options get added, e.g., if the user had their options object as `{someOtherOption: true}`, the code would only need to be changed to this, while keeping the default of `avoidExampleOnConstructors` as `false`.

```js
  const {avoidExampleOnConstructors = false, someOtherOption} = context.options[0] || {};
```

Same with the other block like this below.